src/modelmanager.cpp

```diff
@@ -322,23 +327,54 @@ Status ModelManager::loadModelsConfig(rapidjson::Document& configJson) {
         return StatusCode::JSON_INVALID;
     }
     std::set<std::string> modelsInConfigFile;
-    servedModelConfigs.clear();
+    std::unordered_map<std::string, ModelConfig> newModelConfigs;
     for (const auto& configs : itr->value.GetArray()) {
-        ModelConfig& modelConfig = servedModelConfigs.emplace_back();
+        ModelConfig modelConfig;
         auto status = modelConfig.parseNode(configs["config"]);
+        const auto modelName = modelConfig.getName();
         if (!status.ok()) {
-            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Parsing model:{} config failed",
-                modelConfig.getName());
-            servedModelConfigs.pop_back();
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Parsing model:{} config failed", modelName);
             continue;
         }
-        reloadModelWithVersions(modelConfig);
+        status = reloadModelWithVersions(modelConfig);
         modelsInConfigFile.emplace(modelConfig.getName());
+
+        if (status.ok()) {
+            newModelConfigs.emplace(modelName, std::move(modelConfig));
+        } else if (status == StatusCode::REQUESTED_DYNAMIC_PARAMETERS_ON_SUBSCRIBED_MODEL) {
+            SPDLOG_DEBUG("Will retry to reload model({}) after pipelines are revalidated", modelName);
+            auto it = this->servedModelConfigs.find(modelName);
+            if (it == this->servedModelConfigs.end()) {
+                continue;
+            }
+            this->gatedModelConfigs.emplace_back(std::move(modelConfig));
+            newModelConfigs.emplace(modelName, std::move(it->second));
+            this->servedModelConfigs.erase(modelName);
+        }
```

Comment:
I know that at the moment we have only exit code from reloadModelWIthVersions but maybe at least at SPDLOG_ERROR for other retcodes? Throwing may be idea as well since someone will have to implement such behavior then & decide what to do in such cases.

Reply:
Done